Source/WebKit/UIProcess/Notifications/glib/NotificationService.cpp

```diff
@@ -61,6 +67,48 @@ namespace WebKit {
 
 static const Seconds s_dbusCallTimeout = 20_ms;
 
+#if USE(SKIA)
+
+// Alias to avoid conflicting with write() below.
+ssize_t (*writeToFD)(int, const void*, size_t) = write;
+
+// Simple stream that writes to an fd.
+class fdWStream : public SkWStream {
+public:
+    fdWStream(int fd)
+        : m_fd(fd)
```

Comment:
Give `fdWStream` a virtual destructor and close the fd there. i.e. let the stream object take ownership of the fd.

Or better: use a WTF::UnixFileDescriptor instead of an int. Then you don't even need to write the destructor. And the ownership will be clear. Something like this (untested):

```suggestion
        : m_fd(fd, UnixFileDescriptor::Adopt)
```